rct229/rules/section22/section22rule9.py

```diff
@@ -0,0 +1,131 @@
+            each_rule=Section22Rule9.ChillerFluidLoopRule(),
+            index_rmr="baseline",
+            id="22-9",
+            description="For Baseline chilled water system with cooling capacity of 300 tons or more, the secondary loop shall be modeled with a minimum flow of 25% of the design flow rate.",
+            list_path="ruleset_model_instances[0].fluid_loops[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmr_baseline = context.baseline
+        rmi_b = rmr_baseline["ruleset_model_instances"][0]
+
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list contains all HVAC systems that are modeled in the rmi_b
+        available_type_lists = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+
+        primary_secondary_loop_dictionary = get_primary_secondary_loops_dict(rmi_b)
+
+        return (
+            any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_lists
+                ]
+            )
+            and primary_secondary_loop_dictionary
+        )
+
+    def create_data(self, context, data):
+        rmr_baseline = context.baseline
+        rmi_b = rmr_baseline["ruleset_model_instances"][0]
+
+        chw_loop_capacity_dict = {}
+        for chiller in find_all("chillers[*]", rmi_b):
+            if chiller["cooling_loop"] not in chw_loop_capacity_dict.keys():
+                chw_loop_capacity_dict[chiller["cooling_loop"]] = ZERO.POWER
+            chw_loop_capacity_dict[chiller["cooling_loop"]] += getattr_(
+                chiller, "chiller", "rated_capacity"
+            )
+
+        primary_secondary_loop_dictionary = get_primary_secondary_loops_dict(rmi_b)
+
+        return {
+            "chw_loop_capacity_dict": chw_loop_capacity_dict,
+            "primary_secondary_loop_dictionary": primary_secondary_loop_dictionary,
+        }
+
+    def list_filter(self, context_item, data):
+        fluid_loop_b = context_item.baseline
+        primary_secondary_loop_dictionary = data["primary_secondary_loop_dictionary"]
+        chw_loop_capacity_dict = data["chw_loop_capacity_dict"]
+
+        return (
+            fluid_loop_b["id"] in primary_secondary_loop_dictionary.keys()
+            and chw_loop_capacity_dict[fluid_loop_b["id"]]
+            >= MIN_CHW_PRIMARY_LOOP_COOLING_CAPACITY
+        )
+
+    class ChillerFluidLoopRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section22Rule9.ChillerFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section22Rule9.ChillerFluidLoopRule.SecondaryChildLoopRule(),
+                index_rmr="baseline",
+                list_path="$..child_loops[*]",
```

Comment:
I suggest that a single dot might be better here.

Reply:
Fixed.